src/main/java/org/codehaus/mojo/flatten/DirectDependenciesInheritanceAssembler.java

```diff
@@ -0,0 +1,133 @@
+/**
+ * 
+ */
+package org.codehaus.mojo.flatten;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.enterprise.inject.Specializes;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.apache.maven.model.Model;
+import org.apache.maven.model.ModelBase;
+import org.apache.maven.model.building.ModelBuildingRequest;
+import org.apache.maven.model.building.ModelProblemCollector;
+import org.apache.maven.model.inheritance.DefaultInheritanceAssembler;
+import org.apache.maven.model.merge.MavenModelMerger;
+
+/**
+ * @author kemalsoysal
```

Comment:
at least a short type-comment would be helpful to explain what this class if for. Also a link to issue could be helpful.

Reply:
I hope the type-comment now suffices, I will be happy to describe more if needed